File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.cpp
class Solution {
public:
    bool isdigit(char ch){

        if(ch >='0' && ch<='9'){
            return true;
        }
        return false;
    }
    int myAtoi(string s) {
        
        int n = s.size();
        int i  = 0;
        long long num = 0;
        

        while(i < n && s[i] == ' '){
            i++;
        }
        int sign =1;

        if(i < n && (s[i] == '-'   || s[i] == '+')){
            if(s[i] == '-'){
            sign = -1;
            }

            i++;
        }
        while(i<n){

            if(isdigit(s[i])){
                int digit = s[i] - '0';
                num = num*10 + digit;

                if((sign * num) < INT_MIN) return INT_MIN;
                if ((sign*num) >INT_MAX) return INT_MAX;

            }else{
                    break;
            }
            i++;
        }

        
        
         if((sign * num) < INT_MIN) return INT_MIN;
        if ((sign*num) >INT_MAX) return INT_MAX;

            return sign * num;
    }
};

File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.py
class Solution(object):
    def myAtoi(self, s):
        s = s.strip()
        if not s:
            return 0

        isNegative = False
        i = 0

        if i < len(s) and (s[i] == '-' or s[i] == '+'):
            isNegative = s[i] == '-'
            i += 1

        num = 0
        while i < len(s) and s[i].isdigit():
            num = num * 10 + int(s[i])
            i += 1

        if isNegative:
            num *= -1

        INT_MIN, INT_MAX = -2**31, 2**31 - 1
        return max(INT_MIN, min(INT_MAX, num))
